docs.md

```diff
@@ -173,13 +173,13 @@ val service = new Service[HttpRequest, HttpResponse] {
 #### A `io.finch.request.RequiredParam` reader makes sure that
 * param is presented in the request (otherwise it throws `ParamNoFound(param)` exception)
 * param is not empty (otherwise it throws `ValidationFailed(param, rule)` exception)
-* param may be converted to a requested type `RequiredIntParam`, `RequiredLongParam` or `RequiredBooleanParam`
-(otherwise it throws `ValidationFailed(param, rule)` exception).
+* param may be converted to a requested type through calling `as[Int]`, `as[Float]`, etc.
+(otherwise it throws a `NotParsed` exception).
 
-#### An `io.finch.request.OptionalParam` returns
-* `Future[Some[A]]` if param is presented in the request and may be converted to a requested type `OptionalIntParam`,
-`OptionalLongParam` or `OptionalBooleanParam`
-* `Future.None` otherwise.
+#### An `io.finch.request.OptionalParam.as[A]` returns
+* `Future[Some[A]]` if param is presented in the request and can be converted to the requested type
+* `Future.None` if the param is missing.
+* `NotParsed` exception if the param is presented, but cannot be converted to the requested type
```

Comment:
@vkostyukov is this a typo? I just picked that up from the existing docs, so I'm not sure.